packages/smooth_app/lib/pages/preferences/user_preferences_dev_mode.dart

```diff
@@ -488,3 +473,94 @@ enum DevModeScanMode {
     throw Exception('Unknown index $index');
   }
 }
+
+class LanguagePickerSetting extends StatelessWidget {
+  const LanguagePickerSetting(
+      {required this.delegate,
+      required this.userPreferences,
+      required this.appLocalizations});
+
+  final LocalizationsDelegate<MaterialLocalizations> delegate;
+  final UserPreferences userPreferences;
+  final AppLocalizations appLocalizations;
+
+  @override
+  Widget build(BuildContext context) {
+    final List<String> currentLanguageNameCode = getLanguageCodeLangName(
+        userPreferences.appLanguageCode ??
+            Localizations.localeOf(context).toString());
+
+    final List<String> onlySupportedLanguages = languageNameList
+        .where((String lang) =>
+            delegate.isSupported(Locale(getLanguageCodeLangName(lang)[0])))
+        .toList();
+
+    return ListTile(
+      leading: const Icon(Icons.language),
+      title: Text(
+        '${currentLanguageNameCode[1]} (${currentLanguageNameCode[0]})',
```

Comment:
I'm sure we can afford a new object rather than `[1]` and `[0]`.
Even something very simple like
```dart
class TranslatedLanguage {
  final OpenFoodFactsLanguage language;
  String getLabelInEnglish();
  String getLabelInLanguage();
}
```
Please think OOP.

Reply:
I blame Competitive Programming for that.
I would certainly do it this way.